tests/trx_generator/main.cpp

```diff
@@ -95,6 +101,11 @@ int main(int argc, char** argv) {
          return INITIALIZE_FAIL;
       }
 
+      if(vmap.count("peer-endpoint") && peer_endpoint == "localhost") {
+         ilog("Initializing peer-endpoint from localhost to 127.0.0.1");
+         peer_endpoint = "127.0.0.1";
+      }
```

Comment:
Curious why this is needed?

Reply:
I have a guess: `localhost` resolving to `::1` but p2p plugin only listens on IPv4. If I'm right, could consider changing trx_provider.cpp to resolve only IPv4 addresses.

Reply:
That make sense. You will need something like this in `p2p_connection::connect()`:
```
      tcp::resolver r(_p2p_service);
      tcp::resolver::query q(tcp::v4(), _peer_endpoint, std::to_string(_peer_port));
      auto i = r.resolve(q);
      boost::asio::connect(_p2p_socket, i);
```

Reply:
Thanks, both of you.  That was helpful.  @heifner, that worked a charm. Addressed: [6253abc](https://github.com/AntelopeIO/leap/pull/634/commits/6253abcc4958d1dc5757411fa748bd14b94e8d2d)